### ISSUE TITLE
False positive not-an-iterable on partial async iterator

### ISSUE THREAD

**Issue by romanzdk** [pylint-dev/pylint#9439]
### Bug description

```python
import asyncio
import functools


async def some_iterator(range_, text = 'Text'):
	for i in range(range_):
		yield i, text


partial_some_iterator = functools.partial(some_iterator, range_ = 10)


async def main():
	async for item in partial_some_iterator(text = 'Overriden'):
		print(item)


if __name__ == '__main__':
	asyncio.run(main())
```

### Command used

```shell
pylint ../path/to/file.py
```

### Pylint output

```python
pylint_issue.py:14:19: E1133: Non-iterable value partial_some_iterator(text='Overriden') is used in an iterating context (not-an-iterable)
```

### Expected behavior

No such error - code is working normally

### Pylint version

```shell
pylint 3.0.3
astroid 3.0.2
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
```

### OS / Environment

Ubuntu 22.04.2 LTS
